### PR TITLE
Update swagger spec to 2021-10-04

### PR DESCRIPTION
**Pull request by EmmaZhu**
Thanks for contribution! Please go through following checklist before sending PR.

### PR Branch Destination

- For Azurite V3, please send PR to `main` branch.
- For legacy Azurite V2, please send PR to `legacy-dev` branch.

### Always Add Test Cases

Make sure test cases are added to cover the code change.

### Add Change Log

Add change log for the code change in `Upcoming Release` section in `ChangeLog.md`.

### Development Guideline

Please go to CONTRIBUTION.md for steps about setting up development environment and recommended Visual Studio Code extensions.

### CODE REVIEW

File: src/blob/BlobRequestListenerFactory.ts [hunk 1]
```diff
@@ -113,8 +113,7 @@ export default class BlobRequestListenerFactory
         this.extentStore,
         logger,
         loose
-      ),
-      directoryHandler: {} as any
```
Comment (blueww) [inline src/blob/BlobRequestListenerFactory.ts:117]: why add directoryHandler?

File: src/blob/handlers/ContainerHandler.ts [hunk 2]
```diff
@@ -320,11 +321,25 @@ export default class ContainerHandler extends BaseHandler
...
   }
 
+  public async restore(
+    options: Models.ContainerRestoreOptionalParams,
+    context: Context
+  ): Promise<Models.ContainerRestoreResponse> {
+    throw new NotImplementedError(context.contextId!);
+  }
+
+  public async rename(
```
Comment (blueww) [inline src/blob/handlers/ContainerHandler.ts:331]: What's container rename API? not find in https://learn.microsoft.com/en-us/rest/api/storageservices/operations-on-containers

File: src/blob/handlers/ContainerHandler.ts [hunk 3]
```diff
@@ -633,7 +652,7 @@ export default class ContainerHandler extends BaseHandler
       prefix: options.prefix || "",
       marker: options.marker,
       maxResults: options.maxresults,
-      delimiter,
+      // delimiter,
```
Comment (blueww) [inline src/blob/handlers/ContainerHandler.ts:655]: Why "delimiter" is removed?

File: src/blob/handlers/PageBlobHandler.ts [hunk 4]
```diff
@@ -400,7 +400,6 @@ export default class PageBlobHandler extends BaseHandler
   }
 
   public async getPageRangesDiff(
-    prevsnapshot: string,
     options: Models.PageBlobGetPageRangesDiffOptionalParams,
```
Comment (blueww) [inline src/blob/handlers/PageBlobHandler.ts:403]: Why "prevsnapshot" is removed?

File: swagger/blob.md [hunk 5]
```diff
@@ -22,12 +22,45 @@ enum-types: true
...
+   to
+   "Name": {
+   "type": "string"
+   }
+
+9. Make `ApiVersionParameter` parameter from required to optional.
+
+10. Add `x-ms-creation-time` to Blob_Download API responds
+
+11. Only for 2019-02-02, add "", "deleted" to "ListContainersInclude" enum, add "", "tags", "versions", "deletedwithversions", "legalhold", "permissions" to "ListBlobsInclude" enum.
```
Comment (blueww) [inline swagger/blob.md:55]: We might don't need keep the items for 2019-02-02

File: swagger/blob.md [hunk 6]
```diff
@@ -22,12 +22,45 @@ enum-types: true
 
 2. Updated blocklisttype for list blob blocks from required to optional.
```
Comment (blueww) [inline swagger/blob.md:23]: It's better we can just define the modification with directive in md file as table.md.

File: swagger/blob.md (hunk 5 above)
Comment (blueww) [reply swagger/blob.md:55]: We can delete the old API version swagger.

File: swagger/blob.md [hunk 7]
```diff
@@ -22,12 +22,45 @@ enum-types: true
...
+11. Only for 2019-02-02, add "", "deleted" to "ListContainersInclude" enum, add "", "tags", "versions", "deletedwithversions", "legalhold", "permissions" to "ListBlobsInclude" enum.
+
+12. Only for 2021-10-04, add "" to "ListContainersInclude" enum, add "", "permissions" to "ListBlobsInclude" enum.
+
+13. Only for 2021-10-04, add "Premium" to "AccessTierRequired" enum and "AccessTierOptional" enum.
+    Add "Mutable" to "ImmutabilityPolicyMode" at around line #11994
+
+14. Only for 2021-10-04, add spec for: Blob_GetAccountInfoWithHead, Container_GetAccountInfoWithHead and Service_GetAccountInfoWithHead.
+
+15. Only for 2021-10-04, change return code to 202 for service_submitbatch.
```
Comment (blueww) [inline swagger/blob.md:64]: for the change , please indicate both the orignal value and the target value, like :from 200 to 202

File: swagger/blob.md [hunk 8]
```diff
@@ -22,12 +22,45 @@ enum-types: true
 
 2. Updated blocklisttype for list blob blocks from required to optional.
 
-3. Make "Deleted" and "Snapshot" from required to optional for BlobItem model.
+3. Only for 2019-02-02, make "Deleted" and "Snapshot" from required to optional for BlobItem model.
 
-4. Make `ApiVersionParameter` parameter from required to optional.
+4. Only for 2021-10-04, make "Deleted" and "Snapshot" from required to optional for "BlobItemInternal" model from:
```
Comment (blueww) [inline swagger/blob.md:27]: don't need the new API version. The md file should just target the current swagger API version.

File: src/blob/BlobRequestListenerFactory.ts (hunk 1 above)
Comment (EmmaZhu) [reply src/blob/BlobRequestListenerFactory.ts:117]: directoryHandler includes for datalake service, which is not included in current Azurite blob service support scope.

File: src/blob/handlers/ContainerHandler.ts (hunk 3 above)
Comment (EmmaZhu) [reply src/blob/handlers/ContainerHandler.ts:655]: Service doesn't return "delimiter" for the flat listing operation, this value is only in response for listing by hierarchy operation.

File: swagger/blob.md (hunk 7 above)
Comment (EmmaZhu) [reply swagger/blob.md]: Updated

File: swagger/blob.md (hunk 6 above)
Comment (EmmaZhu) [reply swagger/blob.md:23]: It's directly changed in Swagger spec file in the previous version, not sure how the generator handles this.
I can do try this in a later PR.

File: swagger/blob.md [hunk 9]
```diff
@@ -22,12 +22,45 @@ enum-types: true
...
 
-5. Add `x-ms-creation-time` to Blob_Download API responds
+5. Change for 2021-10-04, change "Name" definition in "BlobItemInternal" from:
```
Comment (XiaoningLiu) [inline swagger/blob.md:29]: I'm good with the regeneration if existing UT passes. Do we have plan better maintain the swagger?

File: swagger/blob-storage-2021-10-04.json
Comment (XiaoningLiu) [inline swagger/blob-storage-2021-10-04.json:1]: PR title is 2021-10-02 but this file name is 2021-10-04

File: src/blob/handlers/PageBlobHandler.ts (hunk 4 above)
Comment (EmmaZhu) [reply src/blob/handlers/PageBlobHandler.ts:403]: It's in the PageBlobGetPageRangesDiffOptionalParams definition.

File: src/blob/handlers/ContainerHandler.ts (hunk 2 above)
Comment (EmmaZhu) [reply src/blob/handlers/ContainerHandler.ts:334]: Added comments for it, it's still private in JS SDK. Or should we just remove it?

File: swagger/blob-storage-2021-10-04.json
Comment (EmmaZhu) [reply swagger/blob-storage-2021-10-04.json:1]: Updated, the API version is 2021-10-04.

File: src/blob/handlers/ContainerHandler.ts (hunk 2 above)
Comment (blueww) [reply src/blob/handlers/ContainerHandler.ts:334]: Better remove it from swagger.

File: swagger/blob.md [hunk 10]
```diff
@@ -10,7 +10,7 @@ enable-xml: true
 generate-metadata: false
 license-header: MICROSOFT_MIT_NO_VERSION
 output-folder: ../src/blob/generated
-input-file: blob-storage-2019-02-02.json
+input-file: blob-storage.json
```
Comment (blueww) [inline swagger/blob.md:13]: Better to add API version in file name

File: swagger/blob.md [hunk 11]
```diff
@@ -22,12 +22,41 @@ enum-types: true
...
+   to
+   "Name": {
+   "type": "string"
+   }
+
+9. Make `ApiVersionParameter` parameter from required to optional.
+
+10. Add `x-ms-creation-time` to Blob_Download API responds
+
+11. Add "" to "ListContainersInclude" enum, add "", "permissions" to "ListBlobsInclude" enum.
```
Comment (blueww) [inline swagger/blob.md:53]: Dup with #5

File: swagger/blob.md [hunk 12]
```diff
@@ -22,12 +22,41 @@ enum-types: true
...
+
+12. Add "Premium" to "AccessTierRequired" enum and "AccessTierOptional" enum.
+    Add "Mutable" to "ImmutabilityPolicyMode" at around line #11994
+
+13. Add spec for: Blob_GetAccountInfoWithHead, Container_GetAccountInfoWithHead and Service_GetAccountInfoWithHead.
+
+14. Change return code from '200' to '202' for service_submitbatch.
+
+15. Change "AllowedHeaders" and "ExposedHeaders" to be not required.
```
Comment (blueww) [inline swagger/blob.md:62]: Better "from required to optional."

**Review by blueww — APPROVED** [review]

**Pull request merged**